src/vulkan/device.cc

```diff
@@ -1010,5 +1057,54 @@ VkFormat Device::GetVkFormat(const Format& format) const {
   return ret;
 }
 
+bool Device::IsRequiredSubgroupSizeSupported(const ShaderType type,
+    const uint32_t required_subgroup_size) const {
+
+    bool is_supported = true;
+    VkShaderStageFlagBits stage = {};
+    switch (type) {
+    case kShaderTypeGeometry:
+        stage = VK_SHADER_STAGE_GEOMETRY_BIT;
+        break;
+    case kShaderTypeFragment:
+        stage = VK_SHADER_STAGE_FRAGMENT_BIT;
+        break;
+    case kShaderTypeVertex:
+        stage = VK_SHADER_STAGE_VERTEX_BIT;
+        break;
+    case kShaderTypeTessellationControl:
+        stage = VK_SHADER_STAGE_TESSELLATION_CONTROL_BIT;
+        break;
+    case kShaderTypeTessellationEvaluation:
+        stage = VK_SHADER_STAGE_TESSELLATION_EVALUATION_BIT;
+        break;
+    case kShaderTypeCompute:
+        stage = VK_SHADER_STAGE_COMPUTE_BIT;
+        break;
+    default:
+    case kShaderTypeMulti:
+        is_supported = false;
+        break;
+    }
+    if ((stage & subgroup_size_control_properties_.requiredSubgroupSizeStages) == 0) {
+        is_supported = false;
```

Comment:
These can both just `return false`. Since the if() then returns the `else if()` can just be an `if()`

Reply:
done

Reply:
done